extensions-core/datasketches/src/main/java/org/apache/druid/query/aggregation/datasketches/theta/SketchAggregator.java

```diff
@@ -42,7 +42,7 @@ public SketchAggregator(BaseObjectColumnValueSelector selector, int size)
 
   private void initUnion()
   {
-    union = new SynchronizedUnion((Union) SetOperation.builder().setNominalEntries(size).build(Family.UNION));
+    union = (Union) SetOperation.builder().setNominalEntries(size).build(Family.UNION);
```

Comment:
we call union.update(..) in updateUnion(..) method defined later in the same file which is prone to failure without SynchronizedUnion.
Also, I wouldn't want to put synchronization on that method itself as it gets called in other situations where synchronization isn't necessary.
So, It probably still makes sense to keep SynchronizedUnion class around.

Reply:
I don't see how synchronizing on Union objects would be avoided in those other situations when synchronization is not necessary.
Could you clarify what is error-prone? We obtain exclusive locks in aggregate() and get() in case they are called at the same time.

Reply:
> It probably still makes sense to keep SynchronizedUnion class around

Given the new API, it is impossible to make SynchronizedUnion a subclass of Union as it was before. But making it just a wrapper breaks a lot of other code, so it was way easier to get rid of it altogether.

Reply:
you're right , Now I notice that updateUnion(..) is also called from synchronized block. nvmnd.